File: front-end/src/app/service-models/app/v1/license/company-invoice-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';

export class CompanyInvoiceSM extends CoinManagementServiceModelBase<number> {
    stripeInvoiceId!: string;
    startDateUTC!: Date;
    expiryDateUTC!: Date;
    discountInPercentage!: number;
    actualPaidPrice!: number;
    currency!: string;
    companyLicenseDetailsId!: number;
    stripeCustomerId!: string;
    amountDue!: number;
    amountRemaining!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/License/CompanyInvoiceSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;

namespace Siffrum.Web.Payroll.ServiceModels.v1.License
{
    public class CompanyInvoiceSM : SiffrumPayrollServiceModelBase<int>
    {
        public string StripeInvoiceId { get; set; }
        public DateTime StartDateUTC { get; set; }
        public DateTime ExpiryDateUTC { get; set; }
        public double DiscountInPercentage { get; set; }
        public decimal ActualPaidPrice { get; set; }
        public string Currency { get; set; }
        public int CompanyLicenseDetailsId { get; set; }
        public string StripeCustomerId { get; set; }
        public long AmountDue { get; set; }
        public long AmountRemaining { get; set; }
    }
}
